commit 80e698e2df5ba2124bdeca37f1e589de58a4d514
Author: Kees Cook <kees@kernel.org>
Date:   Tue Oct 3 16:28:53 2023 -0700

    ASoC: soc-dapm: Annotate struct snd_soc_dapm_widget_list with __counted_by
    
    Prepare for the coming implementation by GCC and Clang of the __counted_by
    attribute. Flexible array members annotated with __counted_by can have
    their accesses bounds-checked at run-time via CONFIG_UBSAN_BOUNDS (for
    array indexing) and CONFIG_FORTIFY_SOURCE (for strcpy/memcpy-family
    functions).
    
    As found with Coccinelle[1], add __counted_by for struct snd_soc_dapm_widget_list.
    Additionally, since the element count member must be set before accessing
    the annotated flexible array member, move its initialization earlier.
    
    Cc: Liam Girdwood <lgirdwood@gmail.com>
    Cc: Mark Brown <broonie@kernel.org>
    Cc: Jaroslav Kysela <perex@perex.cz>
    Cc: Takashi Iwai <tiwai@suse.com>
    Cc: "Gustavo A. R. Silva" <gustavoars@kernel.org>
    Cc: alsa-devel@alsa-project.org
    Cc: linux-hardening@vger.kernel.org
    Link: https://github.com/kees/kernel-tools/blob/trunk/coccinelle/examples/counted_by.cocci [1]
    Signed-off-by: Kees Cook <keescook@chromium.org>
    Reviewed-by: "Gustavo A. R. Silva" <gustavoars@kernel.org>
    Link: https://lore.kernel.org/r/20231003232852.work.257-kees@kernel.org
    Signed-off-by: Mark Brown <broonie@kernel.org>

diff --git a/sound/soc/soc-dapm.c b/sound/soc/soc-dapm.c
index 2512aadf95f7..2e3df47c9cf3 100644
--- a/sound/soc/soc-dapm.c
+++ b/sound/soc/soc-dapm.c
@@ -497,8 +497,8 @@ static int dapm_kcontrol_add_widget(struct snd_kcontrol *kcontrol,
 	if (!new_wlist)
 		return -ENOMEM;
 
-	new_wlist->widgets[n - 1] = widget;
 	new_wlist->num_widgets = n;
+	new_wlist->widgets[n - 1] = widget;
 
 	data->wlist = new_wlist;